Foundation/NSURLSession/NSURLSessionTask.swift

```diff
@@ -838,9 +841,14 @@ extension URLSessionTask {
             var data = Data(capacity: bodyData!.length)
             data.append(Data(bytes: bodyData!.bytes, count: bodyData!.length)) 
 
+            #if !os(Android)
             s.delegateQueue.addOperation {
                 completion(data, response, nil)
             }
+            #else
+            // HACK: Operation Queues not working on Android
+            completion(data, response, nil)
+            #endif
```

Comment:
Operation queue is probably the first thing to look at as it is in Foundation. I have the suspicion that the dispatch thread problem is going to be difficult to resolve. I means that you can not make a call to Java from a background thread at the moment. There may have to be other strategies such as a barrier of some sort.

Any hints you can provide here would be greatly appreciated. I can’t even detemine the code path. Should I be looking at the libkqueue or libpwq code for where threads for queues are created and time out??